### PR TITLE
Add canonical link into playframework documentation

### PR DESCRIPTION
**Pull request by wsargent**
Fixes https://github.com/playframework/playframework.com/issues/43.

Uses Link header as described in https://support.google.com/webmasters/answer/139066?hl=en&rd=1#6

New version (with irrelevant headers removed): 

```
➜  ~  http --headers http://localhost:9000/documentation/2.4.x/Installing
HTTP/1.1 200 OK
Link: <https://www.playframework.com/documentation/2.4.3/Installing>; rel="canonical"
```

Old version (with irrelevant headers removed): 

```
➜  ~  http --headers http://localhost:9000/documentation/2.2.x/Installing
HTTP/1.1 200 OK
Link: <https://www.playframework.com/documentation/2.4.3/Installing>; rel="canonical"
```

Old version with no latest version (with irrelevant headers removed): 

```
➜  ~  http --headers http://localhost:9000/documentation/2.2.x/ScalaGlobal
HTTP/1.1 200 OK
```

### CODE REVIEW

File: app/utils/HeadersHelper.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  /**
+   * Returns the canonical URL that shows the latest version of documentation of a page.
+   */
+  def canonical(page: String)(implicit rh: RequestHeader): String = {
+    // https://github.com/playframework/playframework.com/issues/43
+    val (lang, _) = Router.findLanguageAndVersionPath(rh.path)
+    val relativePath = ReverseRouter.latest(lang, page)
+    // May be relative per https://tools.ietf.org/html/rfc6596#section-3
+    relativePath
+  }
```
Comment (jroper) [inline app/utils/HeadersHelper.scala]: The problem with this is that not every page exists in the latest version of the documentation, which means the link could 404.  Giving the search engines a 404 for the canonical version could well be worse than giving them no canonical link.

The documentation controller can and should work out a canonical link, if one exists, based on the data.  This logic will probably go best in the documentation actor.  There already exists logic for working out if a page exists for a particular version.

**Comment by gmethvin** [comment]
Could we add a banner at the top of old documentation pages indicating that either a newer version exists or that doc page no longer exists? If there is no "latest" documentation for that page, that will still be confusing, especially if people get used to Google showing them the latest doc for other pages.

Also, this may be out of the scope here but mentioning anyway: the selector on the side should somehow indicate which versions actually have that page. Currently it will just redirect to the home page if it doesn't exist.

**Comment by wsargent** [comment]
@gmethvin I like those ideas, and I think we should file them as new features.

**Comment by wsargent** [comment]
This PR addresses @jroper comment "The documentation controller can and should work out a canonical link, if one exists, based on the data. This logic will probably go best in the documentation actor. There already exists logic for working out if a page exists for a particular version."

```scala
+  private def canonicalLinkHeader(page: String) = {
 +    val latestVersion = Version.parse(releases.latest.version).get
 +    val queryPageExists = QueryPageExists(None, latestVersion, None, page)
 +    (documentationActor ? queryPageExists).map {
 +      case PageExists(true, _) =>
 +        val canonicalUrl = s"https://www.playframework.com/documentation/$latestVersion/$page"
 +        val link = s"""<$canonicalUrl>; rel="canonical""""
 +        Seq(LINK -> link)
 +      case other =>
 +        Seq.empty
 +    }
 +  }
 +
```

https://github.com/wsargent/playframework.com/commit/4292f00df9bc2a7380fa2c07bec08b08d3ea3942#diff-21a0d85b8636b36b739d87b60e246143R239

**Pull request merged**